test/end-to-end/core.sh

```diff
@@ -184,6 +184,11 @@ os::cmd::expect_success_and_text "cat '${LOG_DIR}/kubectl-with-token.log'" 'Usin
 os::cmd::expect_success_and_text "cat '${LOG_DIR}/kubectl-with-token.log'" 'kubectl-with-token'
 
 echo "[INFO] Testing deployment logs and failing pre and mid hooks ..."
+# test hook selectors
+os::cmd::expect_success 'oc create -f test/fixtures/complete-dc-hooks.yaml'
+wait_for_command 'oc get pods -l openshift.io/deployer-pod.type=hook-pre  | grep -q Completed' $((60*TIME_SEC))
```

Comment:
These tests should be written as follows:

```sh
os::cmd::try_until_text 'oc get pods -l openshift.io/deployer-pod.type=hook-pre --jsonpath=<path to field>' 'completed'
```

This way we:
1. are consistent in using `os:;cmd`
2. get jUnit integration
3. don't depend on unstable `oc get